<launch>
    <arg name="camera_info_topic" default="/camera/rgb/camera_info"/>
    <arg name="depth_topic" default="/camera/depth_registered/image_raw"/>
    <arg name="target_frame_id" default="map"/>
    <arg name="source_frame_id" default="camera_frame"/>
    <arg name="debug" default="false"/>
    <arg name="rviz" default="false"/>
    <arg name="pose_estimator" default="true"/>

    <node pkg="cutie_ros" type="cutie_service_node.py" name="cutie_service_node" output="screen">
        <param name="/camera_info_topic" type="str" value="$(arg camera_info_topic)"/>
        <param name="/depth_topic" type="str" value="$(arg depth_topic)"/>
        <param name="/target_frame_id" type="str" value="$(arg target_frame_id)"/>
        <param name="/source_frame_id" type="str" value="$(arg source_frame_id)"/>
    </node>

    <group if="$(arg pose_estimator)">
        <node pkg="cutie_ros" type="estimate_3d_pose_publisher.py" name="estimate_3d_pose_publisher" output="screen" />
    </group>

    <group if="$(arg debug)">
        <node pkg="cutie_ros" name="lib_tracking" type="lib_tracking.py" output="screen" />
    </group>

    <group if="$(arg rviz)">
        <node pkg="rviz" type="rviz" name="rviz" args="-d $(find cutie_ros)/io/rviz/hsr_visualization.rviz" />
    </group>

</launch>